src/app/item-details/item-details.component.ts

```diff
@@ -10,16 +13,55 @@ import { Component, Input, ChangeDetectionStrategy, OnChanges } from '@angular/c
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class ItemDetailsComponent implements OnChanges {
-  @Input() item: ActionItem | undefined;
+  @Input() item?: ActionItem;
   @Input() id: string;
-  @Input() errorMessage: string | undefined;
+  @Input() errorMessage?: string;
   @Input() isLoadingActionItem: boolean;
+  @Output() refreshViewAfterDeletion = new EventEmitter();
+  @Output() refreshViewAfterEditing = new EventEmitter();
   daysLeftVisibility: boolean;
 
+  constructor(private matDialog: MatDialog, private tasksService: TasksService) {}
+
   ngOnChanges() {
     if (this.item) {
       const itemDueDate = this.item.dueDate;
       this.daysLeftVisibility = !!itemDueDate;
     }
   }
+  openConfirmationDialog(): void {
+    const dialogRef = this.matDialog.open(ConfirmationDialogComponent, {
+      width: '380px',
+      autoFocus: false
+    });
+
+    dialogRef.afterClosed().subscribe(res => {
+      if (res) {
+        this.tasksService.deleteActionItem(this.id);
+        this.refreshViewAfterDeletion.emit();
+      }
+    });
+  }
+  openEditDialog(): void {
+    const dialogConfig = new MatDialogConfig();
+    dialogConfig.data = {
```

Comment:
**SUGGESTION:** It might be worth to consider to move the default dialog config somewhere (eg. to const) and reuse it here (I mean especially width, height, disableClose etc.)